torch/csrc/distributed/c10d/ddp.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <torch/csrc/utils/tensor_flatten.h>
+
+#include <c10d/ProcessGroup.hpp>
+
+#include <ATen/ATen.h>
+
+#include <cstddef>
+#include <vector>
+
+namespace c10d {
+inline void distBroadcastCoalesced(
+    std::vector<at::Tensor>& tensors,
+    int64_t bufferSize,
+    ProcessGroup& processGroup) {
+  for (auto& group : torch::utils::take_tensors(tensors, bufferSize)) {
+    std::vector<at::Tensor> flatTensor = {
+        torch::utils::flatten_dense_tensors(group.tensors)};
+    BroadcastOptions broadcastOptions;
+    broadcastOptions.rootRank = 0;
+    broadcastOptions.rootTensor = 0;
+    processGroup.broadcast(flatTensor, broadcastOptions)->wait();
```

Comment:
It might actually be a better idea to first queue all the waits, and only then trigger the copies (you can accumulate the futures in a vector).

Reply:
Good idea, I'll do that

Reply:
This should be a decent improvement since chunks are now broadcast in parallel, while the code used to be all serial?